enclave-runtime/src/global_components.rs

```diff
@@ -0,0 +1,54 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::ocall::OcallApi;
+use itc_parentchain::{
+	block_import_dispatcher::immediate_dispatcher::ImmediateDispatcher,
+	block_importer::ParentchainBlockImporter, indirect_calls_executor::IndirectCallsExecutor,
+	light_client::ValidatorAccessor,
+};
+use itp_component_container::ComponentContainer;
+use itp_extrinsics_factory::ExtrinsicsFactory;
+use itp_nonce_cache::NonceCache;
+use itp_stf_executor::executor::StfExecutor;
+use itp_stf_state_handler::GlobalFileStateHandler;
+use itp_types::Block;
+use sgx_crypto_helper::rsa3072::Rsa3072KeyPair;
+use sgx_externalities::SgxExternalities;
+use sp_core::ed25519::Pair;
+
```

Comment:
This is a new file inside the `enclave-runtime` where I plan to define all the concrete types that the enclave uses (keep the crates themselves as generic as possible). In addition, I also keep the global instances of components here to have them constructed/initialized once.